grails-app/conf/BootStrap.groovy

```diff
@@ -267,6 +270,36 @@ class BootStrap {
                 lotAndExpiryControl : product.lotAndExpiryControl,
                 active              : product.active,
                 translatedName      : product.translatedName,
+                glAccount           : product.glAccount,
+                productCatalogs     : product.productCatalogs,
+                productFamily       : product.productFamily,
+            ]
+        }
+
+        JSON.registerObjectMarshaller(ProductCatalog) { ProductCatalog productCatalog ->
+            [
+                    id          : productCatalog.id,
+                    code        : productCatalog.code,
+                    name        : productCatalog.name,
+                    description : productCatalog.description,
+            ]
+        }
+
+        JSON.registerObjectMarshaller(ProductGroup) { ProductGroup productGroup ->
+            [
+                    id          : productGroup.id,
+                    name        : productGroup.name,
+                    description : productGroup.description,
+            ]
+        }
+
+        JSON.registerObjectMarshaller(GlAccount) { GlAccount glAccount ->
+            [
+                id              : glAccount.id,
+                name            : glAccount.name,
+                code            : glAccount.code,
+                description     : glAccount.description,
+                glAccountType   : glAccount.glAccountType.glAccountTypeCode.name(),
```

Comment:
Yes, you have a good point, I think I'll just add a new Marshaler for `GlAccountType`
```
JSON.registerObjectMarshaller(GlAccountType) { GlAccountType glAccountType ->
    return [
        id : glAccountType.id,
        name : glAccountType.name,
        code : glAccountType.code
        accountTypeCode : glAccountType.glAccountTypeCode.name(),
    ]
}
```

Reply:
There is one more thing that occurred to me right now, but I guess  we'll talk about it on the tech huddle.
I added a bunch of new marshallers to the `Product` domain which increased the size of the Product object. What I am afraid of is that I am introducing an issue of over-fetching in some places like the [product search](https://github.com/openboxes/openboxes/blob/develop/grails-app/controllers/org/pih/warehouse/api/ProductApiController.groovy#L101) which returns a list of **Product** instances when `availableItems = true`.
Do you think that this might be a problem ? Because if so, we can look into implementing a separate Product DTO for when `availableItems = true`.